src/engine/BMGame.php

```diff
@@ -3266,27 +3081,26 @@ protected function get_playerDataArray($requestingPlayerIdx) {
         $playerDataArray = array();
 
         // helper arrays of data that BMGame naturally produces in array form
-        $roundScoreArray = $this->get__roundScoreArray();
         $sideScoreArray = $this->get_sideScoreArray();
         $canStillWinArray = $this->get_canStillWinArray();
 
-        foreach ($this->getBMPlayerProps('playerId') as $playerIdx => $playerId) {
+        foreach ($this->playerArray as $playerIdx => $player) {
             $playerData = array(
-                'playerId'            => $playerId,
+                'playerId'            => $player->playerId,
                 'button'              => $this->get_buttonInfo($playerIdx),
                 'activeDieArray'      => $this->get_activeDieArray($playerIdx, $requestingPlayerIdx),
                 'capturedDieArray'    => $this->get_capturedDieArray($playerIdx),
                 'outOfPlayDieArray'   => $this->get_outOfPlayDieArray($playerIdx),
                 'swingRequestArray'   => $this->get_swingRequestArray($playerIdx, $requestingPlayerIdx),
-                'optRequestArray'     => $this->get_optRequestArray($playerIdx),
-                'prevSwingValueArray' => $this->get_prevSwingValueArray($playerIdx),
-                'prevOptValueArray'   => $this->get_prevOptValueArray($playerIdx),
+                'optRequestArray'     => $player->optRequestArray,
+                'prevSwingValueArray' => $player->prevSwingValueArray,
+                'prevOptValueArray'   => $player->prevOptValueArray,
                 'waitingOnAction'     => $this->get_waitingOnActionArray($playerIdx, $requestingPlayerIdx),
```

Comment:
If you're talking about the get_... accessors, the ones that need to stay in BMGame are those where BMGame actively lies about what's happening. I'll take a closer look, though, to see if any of these are simply passing information directly across.